curated_transformers/models/with_strided_spans.py

```diff
@@ -1,16 +1,24 @@
-from typing import Callable, List, Tuple
+from typing import Callable, List, Tuple, Union
 from functools import partial
+from unittest.mock import sentinel
 from thinc.model import Model
 from thinc.types import Ragged, Floats2d, Ints1d
 
+from .output import TransformerModelOutput
+
+# In case of a single list, each element corresponds to a single document/span.
+# For nested lists, each inner list corresponds to a single document/span.
+RaggedInOutT = Union[List[Ragged], List[List[Ragged]]]
+Floats2dInOutT = Union[List[Floats2d], List[List[Floats2d]]]
+
 
 def build_with_strided_spans_v1(stride=96, window=128):
     return partial(with_strided_spans, stride=stride, window=window)
 
 
 def with_strided_spans(
     layer, *, stride=96, window=128
-) -> Model[List[Ragged], List[Ragged]]:
+) -> Model[List[Ragged], Union[TransformerModelOutput, RaggedInOutT]]:
```

Comment:
This ties in with the other comment. But I don't really understand why we need the unions here. Aren't we always dealing with transformer outputs?

Reply:
As mentioned above, we support inner layers that either return a `TransformerModelOutput` or a (nested) list of `Ragged`s in the current implementation.